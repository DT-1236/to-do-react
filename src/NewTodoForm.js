import React, { Component } from 'react';

class NewTodoForm extends Component {
  constructor(props) {
    super(props);
    this.state = { task: '' };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleSubmit(evt) {
    evt.preventDefault();
    // do something with form data
    //this.props.addATask
    this.props.addATask(this.state);
    this.setState({ task: '' });
  }
  handleChange(evt) {
    // runs on every keystroke event
    this.setState({ task: evt.target.value });
  }
  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label htmlFor="task">New Task:</label>
        <input
          name="task"
          value={this.state.task}
          onChange={this.handleChange}
        />
        <button>Add!</button>
      </form>
    );
  }
}
export default NewTodoForm;
