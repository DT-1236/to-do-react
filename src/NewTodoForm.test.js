import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import NewTodoForm from './NewTodoForm';
//in src/**-test.js

//smokeTest

describe('simple smoking snapshots', function() {
  it('renders without crushing', () => {
    shallow(<NewTodoForm />);
  });
  it('matches snapshot', function() {
    let wrapper = shallow(<NewTodoForm />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});

describe('updating form on change event', () => {
  const wrapper = shallow(<NewTodoForm />);
  let input = wrapper.find('input');
  input.simulate('change', { target: { name: 'task', value: 'test' } });
  input = wrapper.find('input');
  expect(input.props().value).toBe('test');
  expect(wrapper.state().task).toBe('test');
});
