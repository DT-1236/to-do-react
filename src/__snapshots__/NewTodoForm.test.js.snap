// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple smoking snapshots matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="task"
  >
    New Task:
  </label>
  <input
    name="task"
    onChange={[Function]}
    value=""
  />
  <button>
    Add!
  </button>
</form>
`;
